rspace/src/main/scala/coop/rchain/rspace/util/package.scala

```diff
@@ -55,4 +55,24 @@ package object util {
       data = gnat.data.sortBy(_.source.hash.bytes.toArray)
     )
   }
+
+  def veccmp(a: ByteVector, b: ByteVector): Int = {
+    val c = a.length - b.length
+    if (c != 0) {
+      c.toInt
+    } else {
+      for (i <- 0L until a.length) {
+        //indexed access of two ByteVectors can be not fast enough,
```

Comment:
Curious - why is this?  What is "fast enough"?

Reply:
"Fast enough" here just note about possible place of unreasonably expensive computations for such simple operation. ByteVector can contain multiple chunks and indexed access may require additional computations on every call: "index in vector -> chunk + index in chunk". Ideal comparer in such situations should be implemented inside ByteVector class and compare chunks faster. Absolutely perfect comparer may also use AVX or SSE ops.